src/main/java/ecdar/controllers/EcdarController.java

```diff
@@ -642,13 +654,15 @@ private void initializeViewMenu() {
 
         menuBarViewCanvasSplit.getGraphic().setOpacity(1);
         menuBarViewCanvasSplit.setOnAction(event -> {
-            final BooleanProperty isSplit = Ecdar.toggleCanvasSplit();
-            if (isSplit.get()) {
-                Platform.runLater(this::setCanvasModeToSingular);
-                menuBarViewCanvasSplit.setText("Split canvas");
-            } else {
+            Ecdar.toggleCanvasSplit();
+        });
+        Ecdar.isSplitProperty().addListener((observable, oldValue, newValue) -> {
+            if (newValue) {
                 Platform.runLater(this::setCanvasModeToSplit);
                 menuBarViewCanvasSplit.setText("Merge canvases");
```

Comment:
Should ```menuBarViewCanvasSplit.setTest("Merge canvases");``` be moved into ```this::setCanvasModeToSplit```? Also for "split canvas".

Reply:
That is not a bad idea :laughing: I went through a couple of iterations of this and should probably have gone through one more :stuck_out_tongue_winking_eye:

Reply:
Noice. :1st_place_medal: